package-builder/build-manifest.nix

```
@@ -307,12 +307,15 @@ pkgs.runCommandNoCC name
       fi
 
       for prog in ''${bin[@]} ''${sbin[@]} ''${libexec[@]}; do
-        # Start by patching shebangs in executables, making sure to prefer the
-        # build wrapper environment over the "develop" environment. Note that
-        # the paths substituted come from executables found in `$PATH` at the
-        # time that this function is invoked, so we automatically patch with
-        # paths found in $FLOX_ENV.
-        patchShebangs "$prog"
+        # Start by patching shebangs in executables, making sure to prefer
+        # interpreters found in the build wrapper environment over the ones
+        # provided by Nix in the process of performing this evaluation.
+        # Note that the paths substituted by `patchShebangs()` are derived
+        # from PATH as it is invoked, so we prepend to the PATH prior to
+        # invocation to ensure we all substituted values have the best
+        # possible chance of coming from the build wrapper environment.
+        PATH="${build-wrapper-env-package}/sbin:${build-wrapper-env-package}/bin:$PATH" \
```

Should this be be 
```suggestion
        PATH="${develop-copy-env-package}/sbin:${develop-copy-env-package}/bin" \
```

Ie set PATH to only what's in the dev env and avoid it substituting anything else

---

No, because the patchShebangs part happens after the data has been copied into `$out`, so the rewriting of `${develop-copy-env-package}` -> `${build-wrapper-env-package}` has already taken place. Good question though!

---

I'll go ahead and resolve this for now, feel free to comment to reopen.